src/utils/multihash.rs

```diff
@@ -113,3 +213,31 @@ impl<const S: usize> multihash_derive::Hasher for IdentityHasher<S> {
         self.i = 0
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::io::Cursor;
+
+    use super::*;
+    use crate::utils::rand::forest_rng;
+    use rand::RngCore as _;
+
+    #[test]
+    fn test_digest_byte_stream() {
+        use MultihashCode::*;
+
+        let mut bytes = vec![0; 10000];
```

Comment:
if we're doing a bit of a fuzzy tests, perhaps it would make sense to randomize the input size as well.

Reply:
Added 0, 1, 1024